Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -1327,6 +1326,53 @@ bool CSSPropertyParser::consumeLegacyTextOrientation(bool important)
     return true;
 }
 
+static bool isValidAnimationPropertyList(CSSPropertyID property, const CSSValueList& valueList)
+{
+    // If there is more than one <single-transition> in the shorthand, and any of the transitions
+    // has none as the <single-transition-property>, then the declaration is invalid.
+
+    if (property != CSSPropertyTransitionProperty || valueList.length() < 2)
+        return true;
+    for (auto& value : valueList) {
+        if (value->isPrimitiveValue() && downcast<CSSPrimitiveValue>(value.get()).isValueID()
```

Comment:
The `isValueID` check here is unneeded. Further, I think there is a helper that does this operation on a CSSValue, checking if it’s a particular keyword, but I think maybe in another source file.

Reply:
Found the one I think you mean in StyleProperties.cpp and moved it to a shared location.